### PR TITLE
Fix up usages of Optionals

### PR DESCRIPTION
**Pull request by LiamClarkeNZ**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

Fixed a few potentially troublesome patterns with Optionals and .get(), also fixed up some potential NPEs. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [X] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by scholzj** [comment]
This seems to build on top of #5204 - It has the same issues. That makes it blocked by it. I think it will also cause rebase issues after #5204 is merged.

**Comment by LiamClarkeNZ** [comment]
> This seems to build on top of #5204 - It has the same issues. That makes it blocked by it. I think it will also cause rebase issues after #5204 is merged.

Oh cripes, the build failures definitely confirm that, I'll move it to draft until #5204 is finished and merged. Cheers @scholzj :)

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java:93]: Again here my preference would be simply 

```java
return config.entries().stream()
    .filter(configEntry -> configEntry.name().equals(key))
    .findFirst()
    .map(ConfigEntry::isDefault)
    .orElse(false);
```

**Review by tombentley — APPROVED** [review]
These changes LGTM. 

I left a couple of comments (essentially making the same point) about how we format stream computations with multiple stages. While it would be more churn to implement that everywhere in this PR I do think it would help readability and avoid some rather long lines. But better to get @scholzj's opinion before you proceed.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java:3277]: Here and in many other places in the tests ... should we get the volume once, store it into a variable and then call the asserts on it? I know this was written badly already before this PR. But if you decide to improve it maybe it should be done properly and not just fix some IDE warnings?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:2036]: My view is that if it fits on one line it should be on one line. If it is on multiple lines, it is definitely better to have one stage per line.

Comment (LiamClarkeNZ) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:2036]: :+1: Sweet as, will have a go at getting it at one stage per line.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java
Comment (LiamClarkeNZ) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java:3277]: Thanks @scholzj, I'm trying to find the balance between specific changes only (in this case, the optional usages), and changes which lower the signal/noise ratio. I'll review my changes and add additional refactorings like the one you suggested, without (hopefully) going overboard :)

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java
Comment (LiamClarkeNZ) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java:96]: Removed this method as it was only used in unit test.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java
Comment (LiamClarkeNZ) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:229]: Hi @scholzj, I wanted to check that this refactoring like many others in this file), pulling out commonly reused expressions that could be expressed as a single variable, doesn't go too far in your opinion. 

I've held back on further similar refactorings after this file, as I don't want to race ahead without your blessing.

Let me know :)

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java
Comment (LiamClarkeNZ) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:443]: withNew<X> was deprecated.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java
Comment (LiamClarkeNZ) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:1102]: Extracted a very common pattern out to reduce scrolling sideways.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:813]: Is this needed?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:98]: Why are these tests not needed anymore?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java:3277]: Fair point. I should not complain about too big PR one day and ask you to add more stuff next day :-). Feel free to split it into another PRs / create issues to cover it.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:229]: I'm probably fine with this. Although I do not think this was necessarily needed. 

I think there are several aspects to handle:
* Code readability
* Performance / resource consumption (The UT/IT tests now run for 30-40 minutes, so saving time is nice)

My particular concern in this respect are mainly the asserts where we do `stream()` several times in a row and search for exactly the same thing just to assert several different parts. That seems rather wasteful. I guess the getters being called here again and again should be less wasteful (although to be honest I'm just guessing - I'm not really expert on this kind of stuff).

Comment (LiamClarkeNZ) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java:229]: Ah yep, good point on the reusage of stream().

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java
Comment (LiamClarkeNZ) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java:117]: Okay, no worries :)

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java
Comment (LiamClarkeNZ) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java:98]: They were testing a method that was only called by those two tests.

**Review by scholzj — APPROVED** [review]

**Pull request merged**
